### PR TITLE
load IWAD from zip, cosmetic changes

### PR DESCRIPTION
**Pull request by rfomin**
* Add more lump filters.

* Simplify `CheckIWAD` function, load IWAD from zip archive.

### CODE REVIEW

File: src/d_main.c [hunk 1]
```diff
@@ -951,19 +909,29 @@ static boolean FileContainsMaps(const char *filename)
...
+    int spaces = 35 - (strlen(desc) / 2);
+    for (int i = 0; i < spaces; ++i)
     {
-        I_Printf(VB_WARNING,
-                 "Unknown Game Version, may not work\n"); // killough 8/8/98
+        I_PutChar(VB_INFO, ' ');
     }
+    I_Printf(VB_INFO, "%s", desc);
+
+    PrintDivider();
```
Comment (rfomin) [inline src/d_main.c:934]: @fabiangreffrath I'm not sure about printing banners. Here is what it looks like:
```
Woof 14.5.0 (built on May 21 2024)

W_Init: Init WADfiles.
 adding C:\downloads\wads\test\hacx12.zip
 - adding HACX.WAD
 adding c:\develop\woof\build_reldeb\src\autoload\all-all\woofhud.lmp
 adding c:\develop\woof\build_reldeb\src\autoload\hacx.wad\brghtmps.lmp
===========================================================================
                         HACX: Twitch n' Kill
===========================================================================
M_LoadDefaults: Load system defaults.
 default file: c:\develop\woof\build_reldeb\src\woof.cfg
Loading DEH lump from file HACX.WAD
Loading DEH file c:\develop\woof\build_reldeb\src\autoload\hacx.wad\bloodcolor.deh
Savegame directory: c:\develop\woof\build_reldeb\src\savegames\HACX.WAD

M_Init: Init miscellaneous info.
R_Init: Init DOOM refresh daemon - [ ......................... ]
P_Init: Init Playloop state.
I_Init: Setting up machine state.
I_InitController: Initialize game controller.
```

Comment (fabiangreffrath) [reply src/d_main.c:934]: I think prboom+ only prints the banner if it has been modified by dehacked?

Comment (rfomin) [reply src/d_main.c:934]: PrBoom+ doesn't show any banners at all. I don't like banners without proper DOS startup screen emulation.
I am just not sure where to print the IWAD version, in this PR we load IWAD first and then identify it.

Comment (fabiangreffrath) [reply src/d_main.c:934]: Then it was Chocolate Doom probably. 🤷

File: src/d_iwad.c [hunk 2]
```diff
@@ -28,20 +28,20 @@
 #include "m_misc.h"
 
 static const iwad_t iwads[] = {
-    {"doom2.wad",     doom2,      commercial,   "Doom II"                        },
-    {"plutonia.wad",  pack_plut,  commercial,   "Final Doom: Plutonia Experiment"},
-    {"tnt.wad",       pack_tnt,   commercial,   "Final Doom: TNT: Evilution"     },
+    {"doom2.wad",     doom2,      commercial,   "DOOM II: Hell on Earth"         },
+    {"plutonia.wad",  pack_plut,  commercial,   "Final DOOM: Plutonia Experiment"},
+    {"tnt.wad",       pack_tnt,   commercial,   "Final DOOM: TNT - Evilution"    },
```
Comment (fabiangreffrath) [inline src/d_iwad.c:33]: But these strings were taken from prboom+, right? I remember changing one of the colons into a dash years ago.

File: src/d_main.c (hunk 1 above)
Comment (rfomin) [reply src/d_main.c:934]: Chocolate always prints banners (but differently than vanilla) and supports dehacked changes. Boom/MBF/PrBoom have `startup` BEX strings, but I've never seen them used. Anyway, I don't think it's worth supporting the original banners unless our startup log is made in libtextscreen.

File: src/d_iwad.c (hunk 2 above)
Comment (rfomin) [reply src/d_iwad.c:33]: These lines are taken from the original DOS Doom source (I changed some things, like "DOOM 2"->"Final DOOM").

Comment (fabiangreffrath) [reply src/d_iwad.c:33]: I mixed it up with this commit then: 

https://github.com/coelckers/prboom-plus/commit/7e442f25884a8f6a8105a912ec0e031977439672

File: src/d_main.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/d_main.c:934]: > Anyway, I don't think it's worth supporting the original banners unless our startup log is made in libtextscreen.

Agreed.

Comment (rfomin) [reply src/d_main.c:934]: So, should I remove banners?

File: src/d_iwad.c (hunk 2 above)
Comment (rfomin) [reply src/d_iwad.c:33]: Original: https://bitbucket.org/gamesrc-ver-recreation/doom/src/bd6beabfd561613b671fe5b8842e7d373731d2c5/d_main.c#lines-1542

File: src/d_main.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/d_main.c:934]: Let's say I wouldn't miss them.

**Pull request merged**